extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/CompositeExceptionMapper.java

```diff
@@ -0,0 +1,32 @@
+package io.quarkus.resteasy.runtime;
+
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+import javax.ws.rs.ext.Providers;
+
+import io.smallrye.mutiny.CompositeException;
+
+@Provider
+public class CompositeExceptionMapper implements ExceptionMapper<CompositeException> {
+
+    @Context
+    Providers p;
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Response toResponse(CompositeException ex) {
+        Throwable t = ex.getCauses().get(0);
```

Comment:
This feels fragile, as there is no guarantee which exceptions will be first.

If you want to prioritise security exceptions I think you need to scan the list first to look for security exceptions that need special handling.

Reply:
Hi @stuartwdouglas, I actually did not plan to do any specific filtering of the security exceptions in this mapper, as it is meant to be a generic mapper which would work for `CompositeException`.
As far as the OIDC exceptions are concerned, it can only be the OIDC-related exception, I've also confirmed it in the debug mode the other day.
In some other flows, not even related to OIDC, `CompositeException` may also happen I guess (example, the user code throws the exception in the Uni exception recovery block), so this mapper will work there too.
If we have a case where the user wants a cause at index 1 be mapped then they'd just register their own custom `CompositeExceptionMapper` mapper.
I reckon that for Quarkus in general we just need to look at the top cause and review it if there will be a case where it won't be sufficient, for now it seems to be fine.

Do you agree :-) ?

Reply:
@stuartwdouglas I've updated this mapper to check for the 1st non-null cause, but for now I'd still rather prefer it be a generic mapper. Does it work for you ?